### PR TITLE
Activate read the docs search when user does `Command/Ctrl + K`

### PR DESCRIPTION
**Pull request by rashidakanchwala**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description
The search modal now opens when the user does CMD+K from keyboard - https://rashida-test-kedro.readthedocs.io/en/latest/

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Comment by astrojuanlu** [comment]
Btw I opened an issue for `Esc` not closing the modal https://github.com/readthedocs/addons/issues/216

**Review by astrojuanlu — COMMENTED** [review]
I confirm this works, but it's difficult for me to read the JavaScript diff. Also for some reason it seems to be showing against `main` and not `develop`. @rashidakanchwala maybe you'll need to squash all the commits into one.

**Comment by astrojuanlu** [comment]
This will need rebasing.

**Comment by stichbury** [comment]
It would be great to get this in, maybe even to the upcoming release. What is needed @rashidakanchwala @astrojuanlu if anything, to get it complete...is it just the merge conflict resolution?

File: docs/source/conf.py [hunk 1]
```diff
@@ -485,26 +485,26 @@ def autodoc_process_docstring(app, what, name, obj, options, lines):  # noqa: PL
     remove_arrows_in_examples(lines)
 
 
-def env_override(default_appid):
-    build_version = os.getenv("READTHEDOCS_VERSION")
+# def env_override(default_appid):
```
Comment (stichbury) [inline docs/source/conf.py:488]: Is there a reason for this chunk of code to be commented out rather than removed?

**Comment by astrojuanlu** [comment]
I think this only needs a rebase and a cleanup of the JS stuff to minimise the diff if possible (or assistance from someone with JS expertise to give it a look)

File: docs/source/conf.py (hunk 1 above)
Comment (rashidakanchwala) [reply docs/source/conf.py:488]: i think we need to uncomment it before we merge. This was only to get the forked version running.

**Review by astrojuanlu — APPROVED** [review]
Works! 🎉 

Will just need to address some linting issues

**Review by stichbury — APPROVED** [review]
LGTM 🏆

**Comment by astrojuanlu** [comment]
This will need to be rebased on top of `main` @rashidakanchwala

**Comment by rashidakanchwala** [comment]
Creating a new PR against main for this.

**Pull request closed without merging**